src/ClientHandle.cpp

```diff
@@ -1532,6 +1539,27 @@ void cClientHandle::HandleSlotSelected(Int16 a_SlotNum)
 
 
 
+void cClientHandle::HandleSpectate(const AString & a_PlayerUUID)
```

Comment:
Wouldn't this just create for every Packet a new Class for the event?
Also it's harder to find bugs, when you use a Event queue to handle all the code.